### DISCUSSION TITLE
Output of Firestore rules.debug should specify line number

### DISCUSSION THREAD

**Discussion by nwaughachukwuma** [firebase/firebase-js-sdk#6352] [General]
When working with Firestore rules on the emulator and using [debug](https://firebase.google.com/docs/reference/rules/rules.debug) to print security rules language objects, variables and statement results as they are being evaluated by the security rules engine, it's often necessary to know the file name and line number a printed output references in any case where there're several `debug` statements across multiple test files.

Simply put, if I have several `debug` statements as shown below and check the outputs written to `firestore-debug.log`, I should know where each output is coming from - it's sometimes cumbersome when there're so much content in the log and it's easy to get lost looking for `debug` output.

```js
// an example function in a sample firestore.rules file 
match /products/{productId} {
   function isAuthenticated() {
     return request.auth != null;
   }
   function oneOf(primeMembers) {
     let ownerId = debug(resource.data.get('ownerId', ''));

     let owner = get(/databases/$(database)/documents/owner/$(ownerId));
     let primeMember = owner ? (owner.data.uid in primeMembers): false;

     return isAuthenticated() && primeMember;
   }
}
```

Potential output in `firestore-debug.log`

```log
...
file: db.products.test.ts
L60: string_value: "u5jxu95H9xbv"
```

**Comment by hsubox76** [comment]
Hi, if I'm understanding correctly this is a feature request for the emulators. Let me know if I'm wrong and this is related to the JS SDK. Otherwise, the repo for the emulators is https://github.com/firebase/firebase-tools (which encompasses the Firebase CLI). You can create an issue there and label it as a feature request.

**Reply by nwaughachukwuma** [reply]
Ok @hsubox76, thanks!
